drivers/hmis/app/graphql/types/hmis_schema/ce_opportunity.rb

```diff
@@ -39,7 +40,7 @@ def referral
         :referrals,
         scope: Hmis::Ce::Referral.viewable_by(current_user).where.not(status: 'rejected').
           order(created_at: :desc),
-      ).first # there should be at most 1 (enforced in an AR validation but not at the database level)
+      ).first
```

Comment:
Per https://github.com/open-path/Green-River/issues/7571 lets take this opportunity to remove the scope from the referral resolver. I think something like this is sufficient:

```
    def referral
      # TODO(#7395): permissions - ensure that user has permission to view referrals at this project
      # return nil unless current_permission?(permission: :can_view_referrals, entity: project)

      load_ar_association(object, :active_referrals)&.first # move active scope to a new active_referral association
    end
```


My understanding is that should fix the weirdness you were seeing with `candidates_generated_at`.

Reply:
Thanks, I think the sort is maybe even not necessary, with the `has_one :active_referral` and `unique_referral_per_opportunity` AR validator.
https://github.com/greenriver/hmis-warehouse/blob/release-161/drivers/hmis/app/models/hmis/ce/referral.rb#L64-L76

Reply:
Scratch that, sorry, this should return the accepted referral too (which is not considered active)